nextcloud_connector.py

```diff
@@ -0,0 +1,395 @@
+        elif action == 'download_file':
+            download_file = kwargs.get('download_file', None)
+            download_path = kwargs.get('download_path', None)
+            file_name = kwargs.get('file_name', None)
+            container_id = self.get_container_id()
+            if download_file is None or download_path is None or file_name is None:
+                return False, rest_result
+            result = nc.get_file(download_file, download_path)
+            rest_result = ph_rules.vault_add(container=container_id, file_location=download_path, file_name=file_name)
+        elif action == 'list':
+            path = kwargs.get('path', None)
+            depth = kwargs.get('depth', None)
+            if path is None or depth is None:
+                return False, rest_result
+            rest_result = nc.list(path, depth=depth)
+            result = True
+        elif action == 'move':
+            initial_path = kwargs.get('initial_path', None)
+            destination_path = kwargs.get('destination_path', None)
+            if initial_path is None or destination_path is None:
+                return False, rest_result
+            result = nc.move(initial_path, destination_path)
+        elif action == 'file_info':
+            path = kwargs.get('path', None)
+            if path is None:
+                return False, rest_result
+            rest_result = nc.file_info(path)
+            result = True
+        else:
+            return False, rest_result
+        if result is False:
+            return False, rest_result
+        return True, rest_result
+
+    def _handle_upload_file(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        vault_id = param['vault_id']
+        self.save_progress("vault_id: {}".format(vault_id))
+        add_random_string = param['add_random_string']
+        try:
+            success, _, data = ph_rules.vault_info(vault_id=vault_id)
+            if not success:
+                return action_result.set_status(phantom.APP_ERROR, "Not found vault for {} id.".format(vault_id))
+            data = list(data)[0]
+            name = data.get('name')
+            file_path = data.get('path')
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, "There was an issue opening the file: " + str(e))
+        if add_random_string is False:
+            destination_path = param['destination_path'].lstrip('/').rstrip('/') + '/' + name
+        else:
+            destination_path = param['destination_path'].lstrip('/').rstrip('/') + '/' + self._randomstr() + '_' + name
+        result, rest_result = self._make_rest_call(action='upload_file', destination_path=destination_path, file_path=file_path)
+        if result is False:
+            action_result.set_status(phantom.APP_ERROR, "Action failed")
```

Comment:
Shouldn't this be a return statement? It looks like it is setting the status to error but will continue on and say it passed.

Reply:
done